src/Native/Unix/System.Security.Cryptography.Native/CMakeLists.txt

```diff
@@ -7,7 +7,7 @@ add_compile_options(-Wno-cast-align)
 add_compile_options(-Wno-reserved-id-macro)
 add_compile_options(-Wno-documentation)
 
-add_definitions(-DPIC=1)
+add_definitions(-DPIC=1 -DOPENSSL_API_COMPAT=0x10100000L)
```

Comment:
Is the OPENSSL_API_COMPAT definition still necessary? It's a bit tricky to use since the features it guards are not built into the actual library and may not be available. That may be useful for the portable builds if you check for API existence, but it's quite undesirable for the non-portable builds.

Reply:
Ah, nevermind, the OpenSSL build system is quite quirky. Apparently the headers enforce that OPENSSL_API_COMPAT is at least OPENSSL_MIN_API and at the same time it's set to OPENSSL_MIN_API for the actual library build.

Then again, setting this definition has the effect that we won't get warnings for any APIs that get deprecated. I am not sure whether that is desirable or not.

Reply:
The main effect I was going for was not adding in as many compatibility \#defines.   Most of them are guarded behind an OPENSSL_API_COMPAT check, though certainly not all of them were.

Reply:
That's what I originally thought until I noticed that you set it to version 1.1.0 anyway. I tried building on macOS with both OpenSSL 1.0.2o and 1.1.0i and it seems to behave the same without defining OPENSSL_API_COMPAT. (I do still get linker errors with 1.1.0, but they are likely unrelated and caused by CMake incorrectly linking to wrong library.)

Reply:
Ah, I see. Makes sense... though it would probably be nicer to #define it conditionally in the header when the default is lower than 0x10100000L.

Reply:
Apparently that's not easily possible since it's impossible to `#include <openssl/opensslconf.h>` and then redefine it. Oh well, I guess your solution makes most sense now. :)